virtcontainers/sandbox.go

```diff
@@ -1162,6 +1162,12 @@ func (s *Sandbox) stop() error {
 		return err
 	}
 
+	// vm is stopped remove the sandbox shared dir
+	sandboxSharedDir := filepath.Join(kataHostSharedDir, s.id)
```

Comment:
This cannot be done here. `kataHostSharedDir` is specific to the `kata_agent.go` implementation, and should never be used by the generic sandbox implementation.

Reply:
Sry for missing this. This should be done more generic.

Reply:
@jshachm You can introduce an agent interface to cleanup agent specific artifacts after the VM has stopped.

Reply:
But wait, any cleanup specific to an agent implementation should be part of either `stopContainer()` if it's related to a container, or `stopSandbox()` if it's related to the sandbox itself.
I don't see the value (and actually this would imply duplication) of introducing a new method to the agent interface.

Reply:
@sboeuf The kata host directory can only be deleted after the VM has stopped since it is shared with the VM through 9p. So this cannot be part of `stopContainer` or `stopSandbox`.

Reply:
Oh yes good point, then in this case, a `agent.cleanup()` function as you suggested sounds like a good idea.